src/common/Finisher.cc

```diff
@@ -3,6 +3,11 @@
 
 #include "Finisher.h"
 
+#include "common/debug.h"
+
+// re-include our assert to clobber the system one; fix dout:
```

Comment:
Mini-experiment:
I reverted Finisher.cc and Finisher.h to edbd6edb0471d5e61815b29a1c4a50e21f3b8d12, and rebuilt; this causes dout complilation errors, absent with this patch. Unfortunately, I've had less luck tracking down the dependencies-- Cond.h is included by Finisher.h, which includes common/Mutex.h and via ceph_context.h atomic.h is included, so it doesn't seem like that is the cause after all. Suggestions welcomed.